cli/azd/pkg/apphost/aca_ingress.go

```diff
@@ -0,0 +1,172 @@
+	for _, binding := range bindings {
+		bindingKey := "default" // default is for those with no container port
+		if binding.TargetPort != nil {
+			bindingKey = fmt.Sprintf("%d", *binding.TargetPort)
+		}
+		endpointByContainerPort[bindingKey] = append(endpointByContainerPort[bindingKey], binding)
+	}
+
+	endpointByContainerPortProperties := make(map[string]*endpointGroupProperties)
+	for containerPort, bindings := range endpointByContainerPort {
+		props := &endpointGroupProperties{httpOnly: true}
+		for _, binding := range bindings {
+			if binding.TargetPort != nil {
+				props.port = *binding.TargetPort
+			}
+			if binding.External {
+				props.external = true
+			}
+			if binding.Scheme != acaIngressSchemaHttp && binding.Scheme != acaIngressSchemaHttps {
+				props.httpOnly = false
+			}
+			if binding.Transport == acaIngressTransportHttp2 {
+				props.hasHttp2 = true
+			}
+		}
+		endpointByContainerPortProperties[containerPort] = props
+	}
+
+	countExternalGroups := 0
+	for _, props := range endpointByContainerPortProperties {
+		if props.external {
+			countExternalGroups++
+		}
+		if props.external && !props.httpOnly {
+			return nil, fmt.Errorf("External non-HTTP(s) endpoints are not supported")
+		}
+	}
+	if countExternalGroups > 1 {
+		return nil, fmt.Errorf("Multiple external endpoints are not supported")
+	}
+
+	var httpOnlyGroups []string
+	for groupKey, props := range endpointByContainerPortProperties {
+		if props.httpOnly {
+			httpOnlyGroups = append(httpOnlyGroups, groupKey)
+		}
+	}
+
+	var ingress string
+	if len(httpOnlyGroups) == 1 {
+		ingress = httpOnlyGroups[0]
+	}
+
+	if ingress == "" {
+		// We have more than one, pick prefer external one
+		var externalHttpOnly []string
+		for _, groupKey := range httpOnlyGroups {
+			if endpointByContainerPortProperties[groupKey].external {
+				externalHttpOnly = append(externalHttpOnly, groupKey)
+			}
+		}
+
+		if len(externalHttpOnly) == 1 {
+			ingress = externalHttpOnly[0]
+		} else if len(httpOnlyGroups) > 1 {
+			return nil, fmt.Errorf("Multiple internal only HTTP(s) endpoints are not supported")
+		}
+	}
+
+	additionalPortsCount := len(endpointByContainerPort)
+	if ingress != "" {
+		additionalPortsCount--
+	}
+	if additionalPortsCount > 5 {
+		return nil, fmt.Errorf("More than 5 additional ports are not supported. " +
+			"See https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview#tcp for more details.")
+	}
+
+	finalIngress := &genContainerAppIngress{}
+
+	if ingress != "" {
+		props := endpointByContainerPortProperties[ingress]
+
+		finalIngress.External = props.external
+		finalIngress.TargetPort = defaultIngressPort
+		finalIngress.Transport = acaIngressSchemaHttp
+		if props.hasHttp2 {
+			finalIngress.Transport = acaIngressTransportHttp2
+		}
+		if props.hasHttp2 || !props.external {
+			finalIngress.AllowInsecure = true
+		}
+
+	} else {
+		port := 0
+		for groupKey := range endpointByContainerPort {
```

Comment:
Because enumerating a go map does not have a fixed order, in practice this is just picking a random entry to use as the default ingress rule and then the rest will be used as additional rules.  Is that what we want?  I'm not sure.

Reply:
I can image that it doesn't matter - but a helpful comment about it why it's okay to just pick one at random and then treat the rest as additional ports would be good?  I feel like in practice this might only be able to happen if every endpoint is TCP only (because otherwise we'd try to make one of the HTTP/HTTPs the default endpoint so we could use the ACA ingress controller?)